eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala

```diff
@@ -38,8 +38,25 @@ object LocalChannelKeyManager {
 }
 
 /**
- * This class manages channel secrets and private keys.
- * It exports points and public keys, and provides signing methods
+ * An implementation of [[ChannelKeyManager]] that supports deterministic derivation of keys, based on the initial
+ * funding pubkey.
+ *
+ * Specifically, there are two paths both of length 8 (256 bits):
+ *   - `fundingKeyPath`: chosen at random using `newFundingKeyPath()`
+ *   - `channelKeyPath`: sha(fundingPubkey(0)) using `ChannelKeyManager.keyPath()`
+ *
+ * The resulting paths looks like so on mainnet:
+ * {{{
+ *  funding:
+ *   - initial funding tx (*):
+ *     47' / 1' / <fundingKeyPath> / <1' or 0'> / 0'
+ *   - splice funding txs (index > 0):
+ *     47' / 1' / <fundingKeyPath> / <1' or 0'> / 6' / <index>
+ *   (*) The initial funding tx is computed differently for historical reasons.
+ *
+ *  all others (payment, revocation, htlc, etc.):
+ *     47' / 1' / <channelKeyPath> / <1'-5'> / <index>
```

Comment:
nit: we should make it obvious that we're hardening the indez:

```suggestion
 *     47' / 1' / <channelKeyPath> / <1'-5'> / <index'>
```

Reply:
That was wrong actually, there is no index, removed in a38095a5f.